src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7180,13 +7216,40 @@ private TypeSyntax ParseType(ParseTypeMode mode = ParseTypeMode.Normal)
                     readonlyKeyword = this.CheckFeatureAvailability(readonlyKeyword, MessageID.IDS_FeatureReadOnlyReferences);
                 }
 
+                SyntaxToken scopedKeyword = null;
+                if (this.CurrentToken.ContextualKind == SyntaxKind.ScopedKeyword)
+                {
+                    var resetPoint = this.GetResetPoint();
+
+                    this.EatToken();
+                    // PROTOTYPE: Can we really assume that the type is followed by an identifier?
+                    bool shouldTreatAsModifier = ScanType() != ScanTypeFlags.NotType && this.CurrentToken.Kind == SyntaxKind.IdentifierToken;
```

Comment:
I think it would be good to lean on the assumption that `scoped` is always a modifier when following `ref` for now, and parse this in a similar way that we parse the ReadOnlyKeyword above. If we end up needing to relax the rules to allow types to be named `scoped` in C# 11, we can make any necessary adjustments at that time.